draft-ietf-privacypass-batched-tokens.md

```diff
@@ -317,15 +326,128 @@ struct {
 ~~~
 
 If the FinalizeBatch function fails, the Client aborts the protocol. Token
-verification works exactly as specified in {{ISSUANCE}}.
+verification works exactly as specified in {{RFC9578}}.
+
+# Arbitrary Batched Token Issuance
+
+This section describes an issuance protocol mechanism for issuing multiple tokens in one round trip between Client and Issuer. An arbitrary batched token request can contain token requests for any token type.
+
+## Client-to-Issuer Request {#arbitrary-client-to-issuer-request}
+
+The Client first creates all TokenRequest it wants to batch. To do so, the client follows
+protocol describing issuance, such as {{RFC9578, Section 5.1}} or {{RFC9578, Section 6.1}}.
+
+The Client then creates a BatchedTokenRequest structured as follows:
+
+~~~tls
+struct {
+   uint16_t token_type;
+   select (token_type) {
+      case (0x0001): /* Type VOPRF(P-384, SHA-384), RFC 9578 */
+         uint8_t truncated_token_key_id;
+         uint8_t blinded_msg[Ne];
+      case (0x0002): /* Type Blind RSA (2048-bit), RFC 9578 */
+         uint8_t truncated_token_key_id;
+         uint8_t blinded_msg[Nk];
+   }
+} TokenRequest;
+
+struct {
+  TokenRequest token_requests<0..2^16-1>;
+} BatchTokenRequest
+~~~
+
+The structure fields are defined as follows:
+
+- "token_type" is a 2-octet integer. TokenRequest MUST be prefixed with a uint16 "token_type" indicating the token type. The rest of the structure follows based on that type, within the inner opaque token_request attribute. The above definition corresponds to TokenRequest from {{RFC9578}}
+
+- "token_requests" are serialized TokenRequests, in network byte order. The number of token_requests, as a 2-octet integer, is prepended to the serialized TokenRequests. In addition, the 2-octet integer length of each TokenRequest is prepended to the serialized TokenRequests.
+
+
+The Client then generates an HTTP POST request to send to the Issuer Request
+URL, with the BatchTokenRequest as the content. The media type for this request
+is "application/private-token-batch-request". An example request for the Issuer
+Request URL "https://issuer.example.net/request" is shown below.
+
+~~~
+POST /request HTTP/1.1
+Host: issuer.example.net
+Accept: application/private-token-batch-response
```

Comment:
As discussed offline, we need 4 media types: 

 - private-token-privately-verifiable-batch-{request|response}
 - private-token-arbitrary-batch-{request|response}